### PR TITLE
Company permissions

### PR DESCRIPTION
**Pull request by eeintech**
Added permissions to views and templates.

Note the change on the `InvenTreeRoleMixin` class to allow `permission_required` declaration in views.

### CODE REVIEW

File: InvenTree/company/templates/company/index.html [hunk 1]
```diff
@@ -12,12 +12,13 @@
 <h3>{{ title }}</h3>
 <hr>
 
+{% if title == 'Manufacturers' and roles.purchase_order.add or title == 'Suppliers' and roles.purchase_order.add or title == 'Customers' and roles.sales_order.add %}
```
Comment (SchrodingersGat) [inline InvenTree/company/templates/company/index.html:15]: This check will fail if the title has been translated from english to another language.

Comment (SchrodingersGat) [reply InvenTree/company/templates/company/index.html:15]: Perhaps a different context variable needs to be supplied to the template?

Comment (eeintech) [reply InvenTree/company/templates/company/index.html:15]: Good point, I'll add another variable to the context

Comment (eeintech) [reply InvenTree/company/templates/company/index.html:15]: Actually found `pagetype` very suitable :smiley:

**Comment by SchrodingersGat** [comment]
@eeintech thanks again for the contribution 👍

**Pull request merged**